sw/device/tests/sim_dv/uart_tx_rx_test.c

```diff
@@ -92,6 +122,110 @@ static volatile bool uart_irq_tx_empty_fired;
 static volatile bool exp_uart_irq_rx_overflow;
 static volatile bool uart_irq_rx_overflow_fired;
 
+// Convert arrays to integer value. Need to print array values to avoid being
+// optimized, so that TB can overwrite its values
+uint32_t bytes2word(uint8_t *bytes) {
+  for (int i = 0; i < 4; ++i) {
+    LOG_INFO("UART uart_idx_array %d %x", i, bytes[i]);
+  }
+  return bytes[3] << 24 | bytes[2] << 16 | bytes[1] << 8 | bytes[0];
+}
+
+// TODO, remove this once pinout configuration is provided
+static void init_uart_pins(uint32_t rx_pad, uint32_t rx_uart, uint32_t tx_pad,
+                           uint32_t tx_uart) {
+  mmio_region_t reg32 = mmio_region_from_addr(
+      PINMUX0_BASE_ADDR + PINMUX_MIO_PERIPH_INSEL_0_REG_OFFSET);
+  uint32_t reg_value = rx_pad;
+  // We've got one insel configuration field per register. Hence, we have to
+  // convert the enumeration index into a byte address using << 2.
+  uint32_t reg_offset = rx_uart << 2;
+  uint32_t mask = PINMUX_MIO_PERIPH_INSEL_0_IN_0_MASK;
+  mmio_region_write32(reg32, reg_offset, reg_value & mask);
+
+  reg32 =
+      mmio_region_from_addr(PINMUX0_BASE_ADDR + PINMUX_MIO_OUTSEL_0_REG_OFFSET);
+  reg_value = tx_uart;
+  // We've got one insel configuration field per register. Hence, we have to
+  // convert the enumeration index into a byte address using << 2.
+  reg_offset = tx_pad << 2;
+  mask = PINMUX_MIO_OUTSEL_0_OUT_0_MASK;
+  mmio_region_write32(reg32, reg_offset, reg_value & mask);
+}
+
+static void init_all_uart_pins() {
+  init_uart_pins(
+      kTopEarlgreyPinmuxInselIoc10, kTopEarlgreyPinmuxPeripheralInUart0Rx,
+      kTopEarlgreyPinmuxMioOutIoc11, kTopEarlgreyPinmuxOutselUart0Tx);
+  init_uart_pins(kTopEarlgreyPinmuxInselIor5,
+                 kTopEarlgreyPinmuxPeripheralInUart1Rx,
+                 kTopEarlgreyPinmuxMioOutIor6, kTopEarlgreyPinmuxOutselUart1Tx);
+  init_uart_pins(
+      kTopEarlgreyPinmuxInselIor7, kTopEarlgreyPinmuxPeripheralInUart2Rx,
+      kTopEarlgreyPinmuxMioOutIor10, kTopEarlgreyPinmuxOutselUart2Tx);
+  init_uart_pins(
+      kTopEarlgreyPinmuxInselIor11, kTopEarlgreyPinmuxPeripheralInUart3Rx,
+      kTopEarlgreyPinmuxMioOutIor12, kTopEarlgreyPinmuxOutselUart3Tx);
+}
+
+void update_uart_base_addr_and_irq_id(uint32_t uart_idx) {
```

Comment:
IMO no need to have an arg - directly use the static const variable.

Reply:
Updated, thanks